cpp/dolfinx/fem/interpolate.h

```diff
@@ -38,6 +38,12 @@ class Function;
 std::vector<double> interpolation_coords(const fem::FiniteElement& element,
                                          const mesh::Mesh& mesh,
                                          std::span<const std::int32_t> cells);
+
+/// Helper type for the data that can be cached to speed up repeated
+/// interpolation of discrete functions across different meshes
+using M2MInterpolationData = decltype(std::function{
```

Comment:
@garth-wells for comments on this style, perhaps its more transparent to type the return type explicitly?

In any case, we normally use `blah_blah_t` for user-defined types.